### PR TITLE
ext_manifest: Use ext manifest by default in output image

### PR DESCRIPTION
**Pull request by ktrzcinx**
Introduced changes add extended manifest at the begin of firmware
image, as a firmware metadata description.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

This PR shouldn't be merged before a flag day.
Added as PR to make it easy to cherry-pick for development purposes.

### CODE REVIEW

**Comment by keyonjie** [comment]
shall we bump the major version for this also?

**Comment by ktrzcinx** [comment]
@keyonjie major abi number should be untouched, moreover this patch is going to hang for a long time.

**Comment by keyonjie** [comment]
> @keyonjie major abi number should be untouched, moreover this patch is going to hang for a long time.

Can you add prefix like [DO NOT MERGE] to the PR subject, otherwise it could be merged by mistake sometimes.

And remove the prefix once it is time for merging it.

**Comment by ktrzcinx** [comment]
@lgirdwood @plbossart @kv2019i @lyakh, is it right time, after v1.6 release, to merge this PR and include extended_manifest to each firmware image by default?

**Comment by kv2019i** [comment]
> 
> 
@ktrzcinx wrote:
> @lgirdwood @plbossart @kv2019i @lyakh, is it right time, after v1.6 release, to merge this PR and include extended_manifest to each firmware image by default?

Yes please! Ext-manifest should be enabled for all V1.7 binaries, so to get maximum testing coverage, we should change the default as soon as possible after 1.6 is done and the "merge window" for V1.7 opens. In kernel, ext-manifest support is in v5.8 and newer kernels and by the time we release 1.7, 5.10 is probably out.

**Comment by lgirdwood** [comment]
@kv2019i just to clarify - is kernel ready if we merge this now ? v1.6 is now on a stable branch.

**Comment by wwittbrx** [comment]
@lgirdwood Please hold off with merging this PR, there is a TGL load error in smoke tests that needs to be resolved first. I will investigate as soon as possible and get back to you.

**Comment by kv2019i** [comment]
@lgirdwood wrote:
> @kv2019i just to clarify - is kernel ready if we merge this now ? v1.6 is now on a stable branch.

Well that's the thing. It was ready many months ago, but as this has not been tested by anyone and has been disabled in FW builds, nobody has been testing this. But yes, support is there in place, so if CI is good, go ahead and merge. The sooner we discover any lurking issues, the better.

**Comment by wwittbrx** [comment]
@lgirdwood CI issues have been resolved, and the PR is good to merge from my perspective.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
